src/main/java/yapper/main/Yapper.java

```diff
@@ -0,0 +1,191 @@
+
+import yapper.command.CommandType;
+import yapper.command.Parser;
+import yapper.exception.EmptyDescriptionException;
+import yapper.exception.InvalidTaskNumberException;
+import yapper.exception.UnknownCommandException;
+import yapper.exception.YapperException;
+import yapper.storage.Storage;
+import yapper.task.Deadline;
+import yapper.task.Event;
+import yapper.task.Task;
+import yapper.task.TaskList;
+import yapper.task.Todo;
+import yapper.ui.Ui;
+
+    /**
+     * The Yapper class serves as the main entry point for the application.
+     * It manages user input, processes commands, and interacts with the task list,
+     * user interface, and storage components to provide the full functionality
+     * of the Yapper task management application.
+     */
+public class Yapper {
+    private final Storage storage;
+    private TaskList tasks;
+    private final Ui ui;
+
+    
+    public Yapper(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            ui.printLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the main loop of the Yapper application. This method continuously
+     * reads commands from the user, processes them, and interacts with the
+     * task list and UI. The loop continues until the user issues the 'bye' command.
+     */
+    public void run() {
+        ui.printWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.printLine();
+                CommandType command = Parser.parseCommand(fullCommand);
+
+                switch (command) {
+                case LIST:
+                    ui.printTasks(tasks);
+                    break;
+                case MARK:
+                    handleMark(fullCommand);
+                    break;
+                case UNMARK:
+                    handleUnmark(fullCommand);
+                    break;
+                case TODO:
+                    handleTodo(fullCommand);
+                    break;
+                case DEADLINE:
+                    handleDeadline(fullCommand);
+                    break;
+                case EVENT:
+                    handleEvent(fullCommand);
+                    break;
+                case DELETE:
+                    handleDelete(fullCommand);
+                    break;
+                case BYE:
+                    isExit = true;
+                    ui.printGoodbye();
+                    break;
+                default:
+                    throw new UnknownCommandException();
+                }
+
+                storage.save(tasks.getTasks());
+            } catch (YapperException | IOException e) {
+                ui.printError(e.getMessage());
+            } finally {
+                ui.printLine();
+            }
+        }
+    }
+
```

Comment:
These names make it much easier to tell what the methods do.

But I'd still add JavaDoc comments here if I were you.